Clarify Range Filters section for readability (#1810) (#1851)
moq-wg/moq-transport: draft-ietf-moq-transport.md

There are five Range Filter parameter types, 0x25-0x29, as shown below.

~~~
SUBGROUP_FILTER {
  Type (vi64) = 0x25,
  Length (vi64),
  [SetID (8)],
  [Range (..) ...]
}

OBJECTID_FILTER {
  Type (vi64) = 0x26,
  Length (vi64),
  [SetID (8)],
  [Range (..) ...]
}

PRIORITY_FILTER {
  Type (vi64) = 0x27,
  Length (vi64),
  [SetID (8)],
  [Range (..) ...]
}

OBJECT_PROPERTY_FILTER {
  Type (vi64) = 0x28,
  Length (vi64),
  [SetID (8)],
  [Property Type (vi64)],
  [Range (..) ...]
}

TRACK_PROPERTY_FILTER {
  Type (vi64) = 0x29,
  Length (vi64),
  [SetID (8)],
  [Property Type (vi64)],
  [Range (..) ...]
}

Range {
  Start (vi64),
  [End (vi64)]
}
~~~

Length (vi64) is the byte count of all fields after itself.  When Length
is 0 (used in REQUEST_UPDATE to remove a filter), no further fields are

> present.  The Object Property and Track Property Filters include a

Reviewer: I tried to clarify, but I am not sure what filter is being removed in all cases?  Does specifying OBJECT_PROPERTY_FILTER with Length of 0 remove all Object property filters for all SetIDs?  Is that what we want?

```suggestion
is 0, there is no filter and no further fields are present.  This can
be used in REQUEST_UPDATE to remove a filter. 
The Object Property and Track Property Filters include a
```

> Property Type (vi64) between SetID and the Ranges.

Reviewer 2: Awkward use of "between".  I think it may be fine to say Property Type is after SetID.

> Each Range is an inclusive Start/End pair.  The final End can be omitted

Reviewer 2: "final End" is a little compact?  "End is required in all but the last pair"?

to indicate no upper bound.  An object matches the filter if its value
falls within any Range (i.e., Ranges are OR'd within a filter parameter).

Start is delta encoded from the prior Range's End (or from 0 for the

> first Range), and End is delta encoded from its own Start.  A delta
> that exceeds 2^64-1 MUST be rejected with REQUEST_ERROR / INVALID_FILTER.

Reviewer: ```suggestion
first Range), and End is delta encoded from its own Start.  If adding the delta
would exceed 2^64-1, it MUST be rejected with REQUEST_ERROR / INVALID_FILTER.
```

For example, ranges 3-5 and 10-15 encode as: Start=3, End=2, Start=5, End=5.

> Filter parameters with the same SetID are AND'd; distinct SetIDs are OR'd.

Reviewer: It'd be good to give an overview of the SetID concept in its own section before the framing, because it seems like an afterthought here.

Reviewer 2: Ian to follow up on Set ID clarifications

(only in the Track and Object Property Filters) repeat in any message,
an endpoint MUST reject this with REQUEST_ERROR with error code INVALID_FILTER.

In REQUEST_UPDATE, Length of 0 removes the filter; non-zero replaces it

> entirely.  If a filter parameter is omitted from REQUEST_UPDATE, it is

Reviewer: What about SetID?

There are five Range Filter parameter types, 0x25-0x29, as shown below.

~~~
SUBGROUP_FILTER {
  Type (vi64) = 0x25,
  Length (vi64),
  [SetID (8)],
  [Range (..) ...]
}

OBJECTID_FILTER {
  Type (vi64) = 0x26,
  Length (vi64),
  [SetID (8)],
  [Range (..) ...]
}

PRIORITY_FILTER {
  Type (vi64) = 0x27,
  Length (vi64),
  [SetID (8)],
  [Range (..) ...]
}

OBJECT_PROPERTY_FILTER {
  Type (vi64) = 0x28,
  Length (vi64),
  [SetID (8)],
  [Property Type (vi64)],
  [Range (..) ...]
}

TRACK_PROPERTY_FILTER {
  Type (vi64) = 0x29,
  Length (vi64),
  [SetID (8)],
  [Property Type (vi64)],
  [Range (..) ...]
}

Range {
  Start (vi64),
  [End (vi64)]
}
~~~

Length (vi64) is the byte count of all fields after itself.  When Length
is 0 (used in REQUEST_UPDATE to remove a filter), no further fields are
present.  The Object Property and Track Property Filters include a

> Property Type (vi64) between SetID and the Ranges.

Reviewer 2: ```suggestion
Property Type (vi64) which follows SetID.
```

Each Range is an inclusive Start/End pair.  The final End can be omitted

> to indicate the Range is open-ended.  An object matches the filter if its

Reviewer 2: ```suggestion
Each Range is an inclusive Start/End pair.  End is optional in the last pair; if omitted
it indicates the last Range is open-ended.  An object matches the filter if its
```
